### DISCUSSION TITLE
Codespace Error - Running in recovery mode due to configuration error

### DISCUSSION THREAD

**Discussion by Souvik06** [community/community#45718] [Codespaces] [answered]
### Select Topic Area

Question

### Body

After updating the codespace, it led to only the Image found showing on the set-up page. I waited for a few hours didn't work. So tried restarting the codespace using https://code.cs50.io/restart. After that, I keep on getting this message every time my code-space is getting opened.
I also tried updating the .devcontainer.json using the https://cs50.readthedocs.io/code/ but still not working.
Moreover tried recreating codespaces by deleting previous ones. But nothing worked.
In this recovery mode, none of the cs50 functions is working, i.e., check50, cs50, update50, etc.

This is my codespace URL - https://souvik06-ominous-spoon-xvqpj94v4jjh9q5q.github.dev/

Kindly, help me out.

I'm hereby attaching the error message I'm having after opening VS code & the creation.log file text
![cs50_codespace_error.png](https://user-images.githubusercontent.com/49316831/215336057-7aba5e70-2e20-4d3a-a0bb-8062e65e6568.png)
[creation_log.txt](https://github.com/community/community/files/10530061/creation_log.txt)

**Comment by lawalTheWest** [comment, marked as answer]
> With Codespaces already open.

> Press Cmd + Shift + P (Mac) or Ctrl + Shift + P (Windows) to open the Command Palette.
 You can also find this in the Settings menu (click on the Settings icon in the lower left of your window).

> Type "rebuild" in the search bar.

> Click on "Codespaces: Rebuild Container"

**Reply by zendovyx** [reply]
I have encountered the same issue but the steps you have outlined brings me to the page where it indicates "setting up codespace" and a message saying image found and doesnt load anymore. If I stop and restart the codespace it returns the same error and I will still be in recovery mode.

**Reply by Souvik06** [reply]
> I have encountered the same issue but the steps you have outlined brings me to the page where it indicates "setting up codespace" and a message saying image found and doesnt load anymore. If I stop and restart the codespace it returns the same error and I will still be in recovery mode.

I tried rebuilding with the process mentioned above in the links... After a few tries it started working again... As per the update from the Harvard team, the issue was with GitHub codespace. So, unless GitHub fixes the bug from their end, people will face these issues

**Reply by Baibhav65** [reply]
Brother please help me!!! My cs50 codespace is working fine but whenever I update50 my codespace,it says to rebuild and when I click on rebuild , a blank screen appears and at the top of the screen it says"Setting up your codespace".But time passes and nothing happens.How to fix it?

**Reply by Calvinmd1** [reply]
thank you very much it worked like magic :)

**Reply by SixtoCalvario** [reply]
thank you

**Reply by Trishan0** [reply]
thank you. it worked

**Reply by Kasidate444** [reply]
Thank you so much.

**Reply by Yiqian-Liang** [reply]
what if it shows command 'github.codespaces.rebuildEnvironment' not found

**Reply by rakeshvellaisamy** [reply]
#1

**Reply by jamescampagnola** [reply]
I'm having the same problem.  I tried the 'rebuild' command as you suggested and that hasn't worked either. Can't figure this out or get tech support anywhere.  Please help.  Thanks. I'm dead in the water...

**Reply by sutrotka** [reply]
same here tried this and didnt fix mine either

**Reply by geosawma** [reply]
same for me

**Comment by lawalTheWest** [comment]
Okay. I would suggest you give them a feedback on this as it would be of help in solving future errors on their paths.

**Comment by AnanyaKAjay** [comment]
For anyone else who has trouble with their codespaces going into recovery mode, you may find the CS50 Docs useful : [https://cs50.readthedocs.io/cs50.dev/#creating-a-new-codespace](https://cs50.readthedocs.io/cs50.dev/#creating-a-new-codespace)

**Reply by Uwe789** [reply]
Excellent tip: this, and only this, resolved my "recovery mode" issue! Thank you!!!

**Reply by AnanyaKAjay** [reply]
You are welcome. Glad to hear that the CS50 Docs helped!

**Reply by Paoletta91** [reply]
Does anyone still have problem even using the CS50 Docs tips? My codespace is launched after the rebuilt process but it still is in recovery mode. 
Does anyone has any other tips, please?

**Reply by AnanyaKAjay** [reply]
Did you try creating a new codespace as suggested in the Docs? That worked for me. Alternatively, you could try rebuilding again. If you decide to create a new codespace, it is probably better not to delete your old one since all your files may not be synced.

**Reply by Paoletta91** [reply]
Thank you so much! It worked!

**Reply by AnanyaKAjay** [reply]
My pleasure.
